shell/platform/android/io/flutter/embedding/engine/systemchannels/SettingsChannel.java

```diff
@@ -80,7 +105,31 @@ public void send() {
               + "\n"
               + "platformBrightness: "
               + message.get(PLATFORM_BRIGHTNESS));
-      channel.send(message);
+      final DisplayMetrics metrics = this.displayMetrics;
+      if (Build.VERSION.SDK_INT < 34 || metrics == null) {
+        channel.send(message);
+        return;
+      }
+      final SettingsChannel.SentConfiguration sentConfiguration = new SentConfiguration(metrics);
+      SettingsChannel.SentConfiguration.sentConfigurations.add(sentConfiguration);
+      message.put(CONFIGURATION_GENERATION, sentConfiguration.generationNumber);
+      channel.send(
+          message,
+          new BasicMessageChannel.Reply() {
+            @Override
+            public void reply(Object reply) {
+              final LinkedList<SettingsChannel.SentConfiguration> pastConfigurations =
+                  SettingsChannel.SentConfiguration.sentConfigurations;
+              // Platform channels guarantees FIFO ordering, sentConfiguration
+              // should be either the first element or the second element in the
+              // linked list. Remove older configurations since Flutter ack'd that
+              // it received a more recent version.
+              while (pastConfigurations.size() > 0
+                  && !pastConfigurations.getFirst().equals(sentConfiguration)) {
+                pastConfigurations.remove();
+              }
+            }
+          });
```

Comment:
This makes me wonder: why do we need to do the config ID dance in the first place if you know when flutter has acknowledged receipt. Couldn't `SettingsChannel.getDisplayMetrics` just always return the last DisplayMetrics for which Flutter has acknowledged receipt?

Reply:
Oh, wait. Is the problem there that the acknowledgment is also send async so it would be possible that flutter already knows the latest config, but settings channel hasn't received that acknowledgement yet?

Reply:
Yeah right now this channel's communication is async in both directions.